WcaOnRails/app/models/wcif.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Wcif < ApplicationRecord
+  belongs_to :competition
+
+  # See https://github.com/thewca/worldcubeassociation.org/wiki/wcif
+  def json
+    json_db = JSON.parse(super || "{}")
+    json_db.merge(
+      "formatVersion" => "1.0",
+      "id" => competition.id,
+    )
+  end
+
+  def json=(new_json)
+    json = self.json.merge(new_json)
+    super(JSON.generate(json))
+  end
+
+  def json_hash
```

Comment:
Yeah, maybe checksum would be a better name? The purpose of this is to prevent people accidentally clobbering a competition's WCIF. Consider the following scenario:

1. Jeremy opens a TNoodle to generate scrambles for US Nats.
2. Lars opens a scheduling software to change the schedule for US Nats.
3. Lars saves his changes to the schedule.
4. Jeremy attempts to save his recently generated scrambles. If we allow this action, this will clobber Lars's changes in step 3. To prevent this, we require that Jeremy send the hash/checksum of the WCIF at the time he fetdched it in step 1. This should data loss due to simultaneous editing.